src/libaktualizr/uptane/imagesrepository.cc

```diff
@@ -102,8 +124,32 @@ bool ImagesRepository::verifyTargets(const std::string& targets_raw) {
 }
 
 std::unique_ptr<Uptane::Target> ImagesRepository::getTarget(const Uptane::Target& director_target) {
-  auto it = std::find(targets.targets.cbegin(), targets.targets.cend(), director_target);
-  if (it == targets.targets.cend()) {
+  // Search for the target in the top-level targets metadata.
+  Uptane::Targets toplevel = targets["targets"];
+  auto it = std::find(toplevel.targets.cbegin(), toplevel.targets.cend(), director_target);
+  if (it == toplevel.targets.cend()) {
```

Comment:
To reduce nesting, I think it could benefit from early exit.

e.g:

    if (it != toplevel.targets.cend()) {
      return std_::make_unique<Uptane::Target>(*it);
    }
    ...

Also:

    if (!fnmatch(pattern ...)) {
      continue;
     }

And:

    if (d_it !=delegate.targets.cend()) {
        return std_::make_unique<Uptane::Target>(*d_it);
    }
    ...

Reply:
Done. Good call, it looks much better.